lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -140,8 +140,9 @@ def inline_metadata(catalog, checksum_type)
     environment_path = Pathname.new File.join(Puppet[:environmentpath], catalog.environment, "")
     list_of_resources = catalog.resources.find_all { |res| res.type == "File" }
 
-    # TODO: get property/parameter defaults if entries are nil in the resource
-    # For now they're hard-coded to match the File type.
+    defaults_resource = Puppet::Type.type(:file).new(:name => '/defaults')
```

Comment:
Ah damn, right. I think that means we've come full circle with needing the provider for the agent :).